repos/spack_repo/builtin/packages/go_sh/package.py

```diff
@@ -0,0 +1,57 @@
+# Copyright Spack Project Developers. See COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack_repo.builtin.build_systems.go import GoPackage
+
+from spack.package import *
+
+
+class GoSh(GoPackage):
+    """A shell parser, formatter, and interpreter. Supports POSIX
+    Shell, Bash, and mksh. Requires Go 1.23 or later."""
+
+    homepage = "https://github.com/mvdan/sh"
+    url = "https://github.com/mvdan/sh/archive/refs/tags/v3.12.0.tar.gz"
+
+    maintainers("mcmehrtens")
+    license("BSD-3-Clause", checked_by="mcmehrtens")
+
+    version("3.12.0", sha256="ac15f42feeba55af29bd07698a881deebed1cd07e937effe140d9300e79d5ceb")
+
+    depends_on("go@1.23:", type="build", when="@3.12.0:")
+
+    variant("shfmt", default=True, description="Build and install shfmt")
+    variant("gosh", default=False, description="Build and install gosh")
+    conflicts("~shfmt~gosh", msg="One of shfmt or gosh must be specified")
+
+    sanity_check_is_file = []
+    with when("+shfmt"):
+        sanity_check_is_file.append(join_path("bin", "shfmt"))
+    with when("+gosh"):
+        sanity_check_is_file.append(join_path("bin", "gosh"))
```

Comment:
You cannot make class level code run "lazily" like that, the with body will always execute.

You'll have to do something like

```suggestion
    @property
    def sanity_check_is_file(self):
        files = []
        if self.spec.satisfies("+shfmt"):
            files.append(join_path("bin", "shfmt"))
        if self.spec.satisfies("+gosh"):
            files.append(join_path("bin", "gosh"))
        return files
```

Reply:
Shoot, I don't know how I missed that. Thank you! I applied your suggestion (1b60cd8f89c22e9899ec63744fce1718214dedf9).